sentry-core/src/test/java/io/sentry/core/SentryEnvelopeTest.kt

```diff
@@ -0,0 +1,131 @@
+package io.sentry.core
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.mock
+import java.io.InputStream
+import java.nio.charset.StandardCharsets
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import org.apache.commons.io.IOUtils
+import org.mockito.Mockito.`when`
+
+class SentryEnvelopeTest {
+
+    @Test
+    fun `deserialize sample envelope with event and two attachments`() {
+        val envelopeReader = EnvelopeReader()
+        val testFile = this::class.java.classLoader.getResource("envelope-event-attachment.txt")
+        val stream = testFile!!.openStream()
+        val envelope = envelopeReader.read(stream)
+        assertNotNull(envelope)
+        assertEquals("9ec79c33ec9942ab8353589fcb2e04dc", envelope.header.eventId.toString())
+        assertEquals(3, envelope.items.count())
+        val firstItem = envelope.items.elementAt(0)
+        assertEquals("event", firstItem.header.type)
+        assertEquals("application/json", firstItem.header.contentType)
+        assertEquals(107, firstItem.header.length)
+        assertEquals(107, firstItem.data.size)
+        assertNull(firstItem.header.fileName)
+        val secondItem = envelope.items.elementAt(1)
+        assertEquals("attachment", secondItem.header.type)
+        assertEquals("text/plain", secondItem.header.contentType)
+        assertEquals(61, secondItem.header.length)
+        assertEquals(61, secondItem.data.size)
+        assertEquals("attachment.txt", secondItem.header.fileName)
+        val thirdItem = envelope.items.elementAt(2)
+        assertEquals("attachment", thirdItem.header.type)
+        assertEquals("text/plain", thirdItem.header.contentType)
+        assertEquals(29, thirdItem.header.length)
+        assertEquals(29, thirdItem.data.size)
+        assertEquals("log.txt", thirdItem.header.fileName)
+    }
+
+    @Test
+    fun `when envelope is empty, reader throws illegal argument`() {
+        val envelopeReader = EnvelopeReader()
+        var stream = mock<InputStream>()
+        `when`(stream.read(any())).thenReturn(-1)
```

Comment:
use `whenever` of mockito-kotlin
like `whenever(mock.stringValue()).thenReturn("test")`

Reply:
Wouldn't that match any method? I was trying to match the read overloaf that takes a parameter

Reply:
my sample is just how to use `whenever` which is the port of `when` from mockito to mockito-kotlin

Reply:
`whenever(stream.read(any())).thenReturn(-1)` would be the case

Reply:
Makes sense. Fixed.